plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -311,6 +318,15 @@ func (k *KafkaConsumer) Start(acc telegraf.Accumulator) error {
 			handler := NewConsumerGroupHandler(acc, k.MaxUndeliveredMessages, k.parserFunc, k.Log)
 			handler.MaxMessageLen = k.MaxMessageLen
 			handler.TopicTag = k.TopicTag
+			//if message headers list specified, put it as map to handler
+			MsgHeadersMap := make(map[string]bool, len(k.MsgHeadersAsTags))
+			if len(k.MsgHeadersAsTags) > 0 {
+				for _, header := range k.MsgHeadersAsTags {
+					MsgHeadersMap[header] = true
+				}
+			}
+			handler.MsgHeadersToTags = MsgHeadersMap
```

Comment:
Please use unexported (i.e. starting with a lowercase letter) local variables! This is common practice in Golang...
```suggestion
			msgHeadersMap := make(map[string]bool, len(k.MsgHeadersAsTags))
			if len(k.MsgHeadersAsTags) > 0 {
				for _, header := range k.MsgHeadersAsTags {
					msgHeadersMap[header] = true
				}
			}
			handler.MsgHeadersToTags = msgHeadersMap
```